man/kernel-install.xml

```diff
@@ -161,8 +159,45 @@
 
   <refsect1>
     <title>Environment variables</title>
+
     <para>If <option>--verbose</option> is used, <varname>$KERNEL_INSTALL_VERBOSE=1</varname> will be set for
     the plugins. They may output additional logs in this case.</para>
+
+    <para>If <varname>MACHINE_ID=</varname> is set and not empty, it will be used as <replaceable>MACHINE-ID</replaceable>,
+    overriding any automatic detection attempts, and must be 32 hexadecimal characters.</para>
+
+    <para><varname>KERNEL_INSTALL_MACHINE_ID=</varname> is set for the plugins to the desired <replaceable>MACHINE-ID</replaceable> –
+    either 32 hexadecimal characters or the special value <literal>Default</literal>.</para>
+    <para><varname>KERNEL_INSTALL_BOOT_ROOT=</varname> is set for the plugins to the root directory (mount point, usually) of the hierarchy
+    where boot-loader entries, kernel images, and associated resources should be placed. Can be overriden by setting <varname>BOOT_ROOT=</varname>.</para>
+
+    <para><varname>KERNEL_INSTALL_LAYOUT=legacy|bls-efi|...</varname>
+    is set for the plugins to select the target layout hierarchy – free-form.
+    Defaults to "bls-efi" if <filename>$BOOT/<replaceable>MACHINE-ID</replaceable></filename> exists, or "legacy" otherwise;
+    overriden by setting <varname>layout=</varname> in <filename>install.conf</filename>,
+
+    <variablelist>
+      <varlistentry>
+        <term>legacy</term>
+        <listitem>
+          <para>A non-BLS bootloader system (like GRUB). Outlined for compatibility.</para>
```

Comment:
this is not a useful explanation, as i have no clue what this effectively means

Reply:
With the ellipses and "free-form" in the options I thought I communicated the open-endedness effectively, but I emphasised this now.